src/components/ContentSidebar/DetailsSidebar.js

```diff
@@ -29,89 +37,306 @@ type ExternalProps = {
+    fetchAccessStatsErrorCallback = (e: $AxiosXHR<any>) => {
+        let accessStatsError;
+
+        if (getProp(e, 'status') === UNAUTHORIZED_CODE) {
+            accessStatsError = {
+                error: messages.fileAccessStatsPermissionsError,
+            };
+        } else {
+            accessStatsError = {
+                maskError: {
+                    errorHeader: messages.fileAccessStatsErrorHeaderMessage,
+                    errorSubHeader: messages.defaultErrorMaskSubHeaderMessage,
+                },
+            };
+        }
+
+        this.setState({
+            isLoading: false,
+            accessStats: undefined,
+            accessStatsError,
+        });
+    };
+
+    /**
+     * File access stats fetch success callback
+     *
+     * @private
+     * @param {Object} accessStats - access stats for a file
+     * @return {void}
+     */
+    fetchAccessStatsSuccessCallback = (accessStats: FileAccessStats): void => {
+        this.setState({
+            accessStats,
+            accessStatsError: undefined,
+            isLoading: false,
+        });
+    };
+
+    /**
+     * Fetches the access stats for a file
+     *
+     * @private
+     * @return {void}
+     */
+    fetchAccessStats(): void {
+        const { api }: Props = this.props;
+        const { file }: State = this.state;
+        this.setState({ isLoading: true });
+        api.getFileAccessStatsAPI(false).get({
+            id: file.id,
+            successCallback: this.fetchAccessStatsSuccessCallback,
+            errorCallback: this.fetchAccessStatsErrorCallback,
+        });
+    }
+
+    /**
+     * Sucess callback for classification change
+     *
+     * @private
+     * @return {void}
+     */
+    classifiationChangeSuccessCallback = (file: BoxItem) => {
+        this.setState({
+            file,
+            isLoading: false,
+        });
+    };
+
+    /**
+     * Error callback for classification change
+     *
+     * @private
+     * @return {void}
+     */
+    classifiationChangeErrorCallback = () => {
+        this.setState({
+            isLoading: false,
+        });
+    };
+
+    /**
+     * Refreshes sidebar when classification is changed
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationChange = (): void => {
+        const { file }: State = this.state;
+        const { api }: Props = this.props;
+        this.setState({ isLoading: true });
+        api.getFileAPI().getFile(
+            file.id,
+            this.classifiationChangeSuccessCallback,
+            this.classifiationChangeErrorCallback,
+            {
+                forceFetch: true,
+                updateCache: true,
+                fields: [FIELD_METADATA_CLASSIFICATION],
+            },
+        );
+    };
+
+    /**
+     * Add classification click handler
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationClick = (): void => {
+        const { onClassificationClick = noop }: Props = this.props;
```

Comment:
Should this default prop value live at the class level?

Reply:
yup